apps/audits/api.py

```diff
@@ -222,9 +222,13 @@ def get_queryset(self):
             'id', 'datetime', 'r_detail', 'r_detail_id',
             'r_user', 'r_action', 'r_type'
         )
-        org_q = Q(org_id=Organization.SYSTEM_ID) | Q(org_id=current_org.id)
-        if resource_id:
-            org_q |= Q(org_id='') | Q(org_id=Organization.ROOT_ID)
+
+        org_q = Q()
+        if not current_org.is_root():
+            org_q = Q(org_id=Organization.SYSTEM_ID) | Q(org_id=current_org.id)
+            if resource_id:
+                org_q |= Q(org_id='') | Q(org_id=Organization.ROOT_ID)
+
         with tmp_to_root_org():
             qs1 = self.get_operate_log_qs(fields, limit, org_q, resource_id=resource_id)
             qs2 = self.get_activity_log_qs(fields, limit, org_q, resource_id=resource_id)
```

Comment:
The most notable difference I noticed is that there seems to be an error in the commented line `with tmp_to_root_org():`, where `.get_operate_log_qs()` is being called without its required arguments. The correct implementation should include these parameters:

```python
org_q = Q() | (Q(org_id=Organization.SYSTEM_ID) & q1) | (current_org.id & q2)
```

For further optimizations on this query and others like it, consider implementing caching mechanisms for common operations to reduce database load.

Additionally, checking if `$resource` exists before querying could potentially avoid unnecessary data retrieval when the value of `$resource` is invalid or does not represent a valid relationship.

This is just one instance highlighting how some slight modifications can make substantial improvements in performance and effectiveness.

Let's modify these two queries accordingly to avoid errors and enhance performance.